contents/tutorials/evaluation-runtimes-and-tags.md

```diff
@@ -0,0 +1,335 @@
+- Flags without evaluation tags evaluate for all requests (backward compatibility)
+
+This helps with cost optimization, environment isolation, and better organization of your flags.
+
+## How they work together
+
+These features apply as sequential filters in the evaluation pipeline:
+
+1. **Runtime filter first**: Excludes flags based on the SDK type (client vs server)
+2. **Tag filter second**: Further filters based on environment context
+
+Think of it as a two-stage security checkpoint:
+
+- Runtime checks if you have the right type of ID (client or server)
+- Tags check if you're going to the right destination (production, staging, etc.)
+
+### Example filtering flow
+
+Consider a flag with:
+
+- Runtime: `server`
+- Evaluation tags: `["production", "api"]`
+
+Here's what happens in different scenarios:
+
+| SDK Type | Environment Tags | Result |
+|----------|-----------------|--------|
+| JavaScript Web | `["production", "web"]` | ❌ Blocked by runtime (client SDK) |
+| Node.js | `["staging", "api"]` | ❌ Blocked by tags (no "production") |
+| Node.js | `["production", "api"]` | ✅ Both filters pass |
+| Python | `["production", "backend"]` | ✅ Both filters pass ("production" matches) |
+
+## Setting up both features
+
+### Step 1: Configure the flag in PostHog
+
+When creating or editing a feature flag:
+
+1. **Set the evaluation runtime**:
+   - Find the "Evaluation runtime" selector
+   - Choose `server`, `client`, or `all` (the default is `all`)
+
+2. **Add evaluation tags**:
+   - Click on "Add tags" and create your tag(s)
+   - Click the bolt icon ⚡ to configure evaluation constraints
+   - Select which tags should act as evaluation constraints
+
+### Step 2: Configure your SDKs
+
+Update your SDK initialization to include evaluation environments. Here's how to set it up for different scenarios:
+
+**Production API server (Node.js):**
+
+```javascript
+const posthog = new PostHog('YOUR_API_KEY', {
+    host: 'https://app.posthog.com',
+    evaluation_environments: ['production', 'api', 'backend']
+})
+```
+
+**Production web app (React):**
+
+```jsx
+<PostHogProvider 
+    apiKey='YOUR_API_KEY'
+    options={{
+        api_host: 'https://app.posthog.com',
+        evaluation_environments: ['production', 'web', 'frontend']
+    }}
+>
+    <App />
+</PostHogProvider>
+```
+
+**Staging microservice (Python):**
+
+```python
+posthog = Posthog(
+    'YOUR_API_KEY',
+    host='https://app.posthog.com',
+    evaluation_environments=['staging', 'payments-service', 'backend']
+)
+```
+
+## Common use cases
+
+### 1. Sensitive server-only features with environment isolation
+
+**Scenario**: Database migration flags that should never leak to clients and only run in specific environments.
+
+**Configuration**:
+
+- Runtime: `server`
+- Evaluation tags: `["production", "migration-runner"]`
+
+**Result**: Only server-side SDKs in production with the "migration-runner" tag can evaluate this flag.
```

Comment:
This sounds like it implies an AND relationship, but above it is stated:
> At least one tag must match for the flag to be included

Reply:
you're so right. The relation is OR'd within the array, so I'll update.